example/src/main/scala/example/WebSocketSimpleClient.scala

```diff
@@ -0,0 +1,29 @@
+package example
+
+import zhttp.service.{ChannelFactory, Client, EventLoopGroup}
+import zhttp.socket.{Socket, SocketProtocol, WebSocketFrame}
+import zio._
+import zio.stream.ZStream
+
+object WebSocketSimpleClient extends zio.App {
+  private val env = EventLoopGroup.auto() ++ ChannelFactory.auto
+  val url         = "ws://localhost:8090/subscriptions"
+
+  private val sa = Socket
+    .collect[WebSocketFrame] {
+      case WebSocketFrame.Text("BAZ") => ZStream.succeed(WebSocketFrame.close(1000))
+      case frame                      => ZStream.succeed(frame)
+    }
+    .toSocketApp
+    .onOpen(Socket.succeed(WebSocketFrame.text("FOO")))
+    .onClose(_ => ZIO.unit)
+    .onError(thr => ZIO.die(thr))
+    .withProtocol(SocketProtocol.uri(url))
```

Comment:
In the simple example don't handle so many scenarios. There should just be enough info so that one can atleast get started with it.